### DISCUSSION TITLE
Supported browsers?

### DISCUSSION THREAD

**Discussion by BernardUK** [elan-language/LanguageAndIDE#888] [General]
Elan looks fantastic ... congratulations! BUT on running any program in Firefox just gives me in the Console:
   A Runtime error occured in the Elan code
   ElanRuntimeError@https://elan-lang.org/beta/web-scripts.js:561:11
(also in RISC OS Iris browser), whereas they work fine in Chrome. Is this to be expected?

**Comment by richardpawson** [comment]
Thank you for the kind comment. We have lots of work to do yet!

>> ... they work fine in Chrome. Is this to be expected?

Yes, I'm afraid so. Because of the very recent version of JavaScript we are using, and also the dependency on the 'File API', we have decided that Elan - _for the time being_ - will be guaranteed only for Chrome.  We will shortly be updating the runtime so that if you access the page from another browser you will get a pop-up notice to that effect.

**Reply by BernardUK** [reply]
Okay, understood. You might consider mentioning Chrome on the Elan home page as well.
By the way, you might have noticed in the error message that 'occured' should be 'occurred' (since we're at school!).

**Comment by richardpawson** [comment]
Thanks for the typo correction - it was spelled correctly in the line underneath it! In a similar vein, though, am wondering if we will get any pushback on `isnt`  - though I'm not willing to put an apostrophe in a keyword!

**Reply by BernardUK** [reply]
Apostrophes in keywords would be way too radical. Keywords `is` and `isnt `are a nice touch as `<>` for `not equal` is quirky and `≠` is on nobody's keyboard. Also `=` might look to the early learner to be only for numerical comparisons. Nuff for now.

**Reply by BernardUK** [reply]
Hurrah! Elan runs the Demo programs in Chromium on a Pi 400.

**Reply by richardpawson** [reply]
That's very good to hear. Thanks for confirming it.

**Reply by BernardUK** [reply]
Latest version 0.10.0-Beta appears to run now also in:
-- Firefox 137.0 (Windows 64-bit)
-- Edge 135.0.2179.54 (64-bit)
-- Safari (iPadOS 17.7.6)
-- Chromium 126.0.6478.164 (RaspberryPiOS 11, 32-bit)
though my tests have not been comprehensive.

(Unsurprisingly not on a 2020 Safari (iMac from 2011))

**Comment by richardpawson** [comment]
Thanks for update. I did a demo on a borrowed but up-to-date Mac and the only issues _I noticed_ were with the font, as Safari does not have Consolas - so the thunderflash symbol for a breakpoint didn't show up.
